### DISCUSSION TITLE
Glass UI

### DISCUSSION THREAD

**Discussion by yvt** [helix-editor/helix#3597] [Ideas]
I wonder if there is interest in something like this?

![ss2](https://user-images.githubusercontent.com/5253988/187205706-ac380ecf-2247-4282-865b-ceab05e9beb6.jpeg)

![ss1](https://user-images.githubusercontent.com/5253988/187205693-e286637f-6ed4-42e7-b33f-834a46ce7d6f.jpeg)

![ss4](https://user-images.githubusercontent.com/5253988/187213259-e8717b3c-df2d-48dc-ac78-d3b07538ae55.jpeg)

**Comment by the-mikedavis** [comment]
That looks nice! I'm not sure it's possible to do until there's a GUI though because of terminal limitations, right?

**Reply by CptPotato** [reply]
I'm pretty sure this is in-terminal and uses various block characters (like [these](https://en.wikipedia.org/wiki/Symbols_for_Legacy_Computing)) with bg and fg colors to approximate the blurred gradient as closely as possible.

**Reply by yvt** [reply]
It's certainly possible! The pictures above are actual screenshots from <https://github.com/yvt/helix/tree/⚗/blur>.

**Reply by CptPotato** [reply]
Very cool!
Sadly I can't run the patched version since it doesn't build with MSVC (because of `gmp-mpfr-sys` required by the `srgb` crate).

**Reply by Omnikar** [reply]
I tried it out and it seems really awesome! Although, I got near abysmal performance if I didn't compile in release mode.
